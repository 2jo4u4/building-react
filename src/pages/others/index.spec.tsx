import React from 'react'
import Other from './index'
import { fireEvent, render } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('Test', () => {
  const { queryByLabelText, getByLabelText } = render(<Other labelOn="On" labelOff="Off" />)

  expect(queryByLabelText(/off/i)).toBeTruthy()

  fireEvent.click(getByLabelText(/off/i))

  expect(queryByLabelText(/on/i)).toBeTruthy()

  const component = renderer.create(<Other labelOn="On" labelOff="Off" />)
  const tree = component.toJSON() as renderer.ReactTestRendererJSON
  expect(tree).toMatchSnapshot()
})
