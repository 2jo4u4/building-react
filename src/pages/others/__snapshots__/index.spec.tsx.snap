// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;
